walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -41,19 +41,41 @@ private boolean eventParticipantInMeeting(
   /**
    * Returns a collection of all meeting times that will work for all attendees' schedules.
    *
-   * <p>The open meeting times are determined by iterating through each event and creating a new
-   * {@code TimeRange} for each range between meetings. If the duration is greater than or equal to
-   * the duration of the requested meeting, that {@code TimeRange} instance is added to the list
-   * {@code openMeetingTimes}. The initial value of {@code endOfEarlierEvent} is set to the start of
-   * the day for convenience when evaluating the first event. Also, the end of day time is appended
-   * as an additional event to the current event list to avoid more conditional statements.
+   * <p>Initially, the meeting attendees is set based on {@code considerOptionalAttendees}. Once
+   * this is determined, the event list is filtered and sorted. More specifically, the events are
+   * filtered if it does not contain at least one participant that is also a meeting participant.
+   * The open meeting times are determined by iterating through each event and creating a new {@code
+   * TimeRange} for each range between meetings. If the duration is greater than or equal to the
+   * duration of the requested meeting, that {@code TimeRange} instance is added to the list {@code
+   * openMeetingTimes}. The initial value of {@code endOfEarlierEvent} is set to the start of the
+   * day for convenience when evaluating the first event. Also, the end of day time is appended as
+   * an additional event to the current event list to avoid more conditional statements.
```

Comment:
This description is useful, but the code for the function is pretty short. I feel that a lot of what's stated here is just restating the code. I find these function level comments very useful when they describe the semantics of the code rather than the specifics of it. The specifics are better suited to be written as in-line comments on code that is non-intuitive. If the code is readable on its own, then specifics are often not needed at all.

This is fine, but it may be useful to consider my perspective above in future commits.

Reply:
Coming back to this now after forgetting about it when starting to design SLURP. 

After reading this and my enormous comment I agree that this is pretty much just restating the code so I will just remove it and place some of the details above bits of the code that I do not immediately understand its purpose after coming back and looking at this.